tools/clip/run.py

```diff
@@ -84,71 +76,333 @@ def get_meta_text(lr: LogReader, route: Route):
     f"modified: {str(init_data.dirty).lower()}",
   ])
 
+def populate_car_params(lr: LogReader):
+  init_data = lr.first('initData')
+  assert init_data is not None
+
+  params = Params()
+  for cp in init_data.params.entries:
+    try:
+      params.put(cp.key, params.cpp2python(cp.key, cp.value))
+    except UnknownKeyName:
+      logger.warning(f"unknown Params key '{cp.key}', skipping")
+
+  car_params = lr.first('carParams')
+  if car_params:
+    params.put("CarParams", car_params.as_builder().to_bytes())
+  logger.debug('persisted CarParams')
+
+def patched_update_msgs(self, cur_time, msgs):
```

Comment:
we use this in multiple places now, i think we should just put this in SubMaster directly as a separate mode